web_ui/frontend/components/layout/AuthenticatedContent.tsx

```diff
@@ -1,29 +1,106 @@
-import {Box, BoxProps, Skeleton} from "@mui/material";
-import {useEffect, useState} from "react";
-import {isLoggedIn} from "@/helpers/login";
+"use client"
 
-const AuthenticatedContent = ({...props} : BoxProps) => {
+import {Box, BoxProps, Button, Skeleton, Typography} from "@mui/material";
+import useSWR from "swr";
+import {getUser} from "@/helpers/login";
+import {useEffect, useMemo, useState} from "react";
+import Link from "next/link";
+import {useRouter} from "next/navigation";
+import {User} from "@/index";
 
-    const [authenticated, setAuthenticated] = useState<boolean | undefined>(undefined)
+
+const Circle = ({children}: {children: React.ReactNode}) => {
+    return (
+        <Box display={"flex"} justifyContent={"center"} alignItems={"center"} height={"100%"} width={"100%"}>
+            <Box padding={6} borderRadius={"100%"} bgcolor={"primary.light"} sx={{aspectRatio: 1}}>
+                <Box display={"flex"} justifyContent={"center"} alignItems={"center"} height={"100%"} flexDirection={"column"}>
+                    {children}
+                </Box>
+            </Box>
+        </Box>
+    )
+}
+
+interface AuthenticatedContentProps {
+    promptLogin?: boolean;
+    redirect?: boolean;
+    children: React.ReactNode;
+    boxProps?: BoxProps;
+    checkAuthentication?: (user: User) => boolean;
+}
+
+const AuthenticatedContent = ({promptLogin = false, redirect = false, children, boxProps, checkAuthentication}: AuthenticatedContentProps) => {
+
+    if(redirect && promptLogin){
+        throw new Error("redirect XOR promptLogin must be true")
+    }
+
+    const router = useRouter()
+    const {data, isValidating, isLoading, error} = useSWR("getUser", getUser, { refreshInterval: 1000 * 60, revalidateOnMount: true })
+    const [pageUrl, setPageUrl] = useState<string>("")
+
+    const authenticated = useMemo(() => {
+        if(data && checkAuthentication){
+            return checkAuthentication(data)
+        } else {
+            return data?.authenticated
+        }
+    }, [data, checkAuthentication])
 
     useEffect(() => {
-        (async () => {
-            const loggedIn = await isLoggedIn()
-            setAuthenticated(loggedIn)
-            if(!loggedIn){
-                window.location.replace("/view/login/")
-            }
-        })()
+        const path = window.location.pathname
+        const pathWithoutView = path.replace("/view", "")
+        const pathUrlEncoded = encodeURIComponent(pathWithoutView)
+
+        setPageUrl(pathUrlEncoded)
     }, []);
 
-    if(authenticated === false){
-        return null
+    // Redirect to login page if not authenticated and redirect is true
+    useEffect(() => {
+
+        if(!isValidating && !authenticated && redirect){
+
+            const path = window.location.pathname
+            const pathWithoutView = path.replace("/view", "")
```

Comment:
Same here, can we take query param to redirect user back to where they were once they are authenticated?

Reply:
This should, did you find a case where it didn't?

Reply:
When I hit the button to register an origin at the registry, it redirects me to login page, but after I logged in I was redirected to the registry page

Reply:
Same thing with the config page. I used admin password for auth